File: ui/gfx/size.cc
// Copyright (c) 2012 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "ui/gfx/size.h"

#if defined(OS_WIN)
#include <windows.h>
#elif defined(OS_MACOSX)
#include <ApplicationServices/ApplicationServices.h>
#endif

#include "base/logging.h"
#include "base/stringprintf.h"
#include "ui/gfx/size_base.h"
#include "ui/gfx/size_base_impl.h"

namespace gfx {

template class SizeBase<Size, int>;

Size::Size() : SizeBase<Size, int>(0, 0) {}

Size::Size(int width, int height) : SizeBase<Size, int>(0, 0) {
  set_width(width);
  set_height(height);
}

#if defined(OS_MACOSX)
Size::Size(const CGSize& s) : SizeBase<Size, int>(0, 0) {
  set_width(s.width);
  set_height(s.height);
}

Size& Size::operator=(const CGSize& s) {
  set_width(s.width);
  set_height(s.height);
  return *this;
}
#endif

Size::~Size() {}

#if defined(OS_WIN)
SIZE Size::ToSIZE() const {
  SIZE s;
  s.cx = width();
  s.cy = height();
  return s;
}
#elif defined(OS_MACOSX)
CGSize Size::ToCGSize() const {
  return CGSizeMake(width(), height());
}
#endif

std::string Size::ToString() const {
  return base::StringPrintf("%dx%d", width(), height());
}

}  // namespace gfx

File: src/chrome/android/feed/core/java/src/org/chromium/chrome/browser/feed/library/sharedstream/publicapi/menumeasurer/Size.java
// Copyright 2019 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.chrome.browser.feed.library.sharedstream.publicapi.menumeasurer;

import androidx.annotation.Nullable;

/**
 * Class to represent the width and height of a view in pixels. This is a minimal version of {@link
 * android.util.Size} which is only available in API 21.
 */
public class Size {
    private final int mWidth;
    private final int mHeight;

    public Size(int width, int height) {
        this.mWidth = width;
        this.mHeight = height;
    }

    /** Gets the width of the size in pixels. */
    public int getWidth() {
        return mWidth;
    }

    /** Gets the height of the size in pixels. */
    public int getHeight() {
        return mHeight;
    }

    @Override
    public boolean equals(@Nullable Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof Size)) {
            return false;
        }

        Size size = (Size) o;

        return mWidth == size.mWidth && mHeight == size.mHeight;
    }

    @Override
    public int hashCode() {
        return 31 * mWidth + mHeight;
    }
}
